spec.bs

```diff
@@ -1790,7 +1795,7 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
       <dt>{{GenerateBidInterestGroup/enableBiddingSignalsPrioritization}}
       <dd>|ig|'s [=interest group/enable bidding signals prioritization=]
       <dt>{{GenerateBidInterestGroup/priorityVector}}
-      <dd>|ig|'s [=interest group/priority vector=] if not null, otherwise {{undefined}}
+      <dd>|ig|'s [=interest group/priority vector=] if not null, otherwise not set.

```

Comment:
I don't think "unset" is a valid state. Null seems best, and if that doesn't immediately work then maybe you need two levels of optionality (null vs something to indicate "set vs unset", and then maybe another level of optionality for domain-specific stuff??). I'm not really sure, but I do know that "not set" I don't think is a real thing. Undefined or null is right.

Reply:
Undefined or null is not what we actually do. We don't set the field at all; e.g. ("priorityVector" in igGenerateBid) will evaluate to false.

Reply:
I see. I'm sorry I think I was wrong here, unset should be fine since this is a dictionary and all members are optional anyways. Thanks.

Reply:
Oh shoot, I think my first point was actually right. Based on how WebIDL works for dictionaries, "unset" will simply result in the value being "undefined" for `priorityVector` in this case. There is no way to "unset" it (literally exclude it as a property altogether) given the shape of the WebIDL dictionary object. I think what you'll have to do is one of the following, ranked in my order of preference:

1. Change the implementation to unconditionally set the value to undefined in the case where you currently exclude it altogether (this is more webby and consistent with the web platform elsewhere)
2. Manually `[[Delete]]` the member after creation, before it gets sent back to the web platform. This one is a little sketchy, but it is probably close enough to get the point across and be considered valid. You would do something like this: https://wicg.github.io/turtledove/#:~:text=Perform%20!global.%5B%5BDelete%5D%5D(%22Date%22)

I think (1) is better, is there any possibility of considering it?

Reply:
We're super paranoid about backwards compatibility.

Can I just... not set it? It's optional, after all.

Reply:
WebIDL -> JS conversion seems fine for this?
https://webidl.spec.whatwg.org/#js-dictionary:~:text=If%20V%5Bkey%5D%20exists%2C%20then%3A

Reply:
Yeah looks like I was wrong again, not sure why I thought the members _always_ existed in the corresponding JS object. For what its worth just as extra knowledge here, I did learn that supplying `undefined` has the same result as omitting it completely, or leaving it "unset": https://webidl.spec.whatwg.org/#ref-for-dictionary-member-optional:~:text=In%20the%20JavaScript%20binding%2C%20a,in%20the%20dictionary%20value%20otherwise.

But anyways I think we're all good here, thanks for the patience.